app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -0,0 +1,352 @@
+          variables: <String, dynamic>{
+            'sOwner': owner,
+            'sName': name,
+            'sLabelName': labelName,
+          }),
+    );
+
+    if (result.hasErrors) {
+      log.error(jsonEncode(result.errors));
+      throw const BadRequestException('GraphQL query failed');
+    }
+
+    return result.data;
+  }
+
+  Future<bool> _removeLabel(
+    String owner,
+    String name,
+    String id,
+    String message,
+    String labelId,
+    GraphQLClient client,
+  ) async {
+    final QueryResult result = await client.mutate(MutationOptions(
+      document: _removeLabelMutation,
+      variables: <String, dynamic>{
+        'id': id,
+        'sBody': message,
+        'labelId': labelId,
+      },
+    ));
+    if (result.hasErrors) {
+      log.error(jsonEncode(result.errors));
+      return false;
+    }
+    return true;
+  }
+
+  Future<bool> _mergePullRequest(
+    String owner,
+    String name,
+    String id,
+    String sha,
+    Logging log,
+    GraphQLClient client,
+  ) async {
+    final QueryResult result = await client.mutate(MutationOptions(
+      document: _mergePullRequestMutation,
+      variables: <String, dynamic>{
+        'id': id,
+        'oid': sha,
+      },
+    ));
+
+    if (result.hasErrors) {
+      log.error(jsonEncode(result.errors));
+      return false;
+    }
+    return true;
+  }
+
+  /// Parses a GraphQL query to a list of [_AutoMergeQueryResult]s.
+  ///
+  /// This method will not return null, but may return an empty list.
+  List<_AutoMergeQueryResult> _parseQueryData(Map<String, dynamic> data) {
+    final Map<String, dynamic> repository = data['repository'];
+    if (repository == null || repository.isEmpty) {
+      return <_AutoMergeQueryResult>[];
+    }
+
+    final Map<String, dynamic> label = repository['labels']['nodes'].single;
+    if (label == null || label.isEmpty) {
+      return <_AutoMergeQueryResult>[];
+    }
+    final String labelId = label['id'];
+
+    return label['pullRequests']['nodes']
+        .cast<Map<String, dynamic>>()
+        .map<_AutoMergeQueryResult>((Map<String, dynamic> pullRequest) {
+      final String id = pullRequest['id'];
+      final int number = pullRequest['number'];
+      final bool mergeable = pullRequest['mergeable'] != 'MERGEABLE';
+      final List<Map<String, dynamic>> reviews =
+          pullRequest['reviews']['nodes'].cast<Map<String, dynamic>>();
+      bool hasApproval = false;
+      bool hasChangesRequested = false;
+      for (Map<String, dynamic> review in reviews) {
+        switch (review['state']) {
+          case 'APPROVED':
+            hasApproval = true;
+            break;
+          case 'CHANGES_REQUESTED':
+            hasChangesRequested = true;
+            break;
+        }
+      }
+      final Map<String, dynamic> commit =
+          pullRequest['commits']['nodes'].single['commit'];
+      final String sha = commit['oid'];
+      final bool ciSuccessful = commit['status']['state'] == 'SUCCESS';
```

Comment:
This may not be sufficient. We still need to check that at least a fixed number of tests are passed just in case that some presubmit tests are not started yet: https://github.com/flutter/flutter/issues/43672

Reply:
I see.  Do we know what that number is?

I wonder if this one should be ok because we expect people to only add this label _after_ all other checks have passed.  but if we want to check/enforce that, we'd need to know the number...

Reply:
You'll have the number automatically once https://github.com/flutter/flutter/pull/43681 lands. For now, that number should be 99.

Reply:
You could also explicitly wait an hour after the last commit was uploaded before considering a PR. That would have the advantage of giving time for all the checks to begin, and for people to rereview the PR before it landed. (I would probably do this in addition to the test of how many checks we want.)

If you want something like https://github.com/flutter/flutter/pull/43681 then we should talk about what that should look like. I assume you don't want to check out the entire PR locally to run code from it...

Reply:
We could require one particular shard to always run, and for that shard to include the number you care about in its output? Would reading the data from a shard be possible here?

Reply:
I believe the script here can either clone the PR branch and run the commit locally, or check the shard output through either Cirrus or Github GraphQL API.

FWIW, waiting 1 hour alone may still be insufficient. Cirrus has run into scheduling problems before so it's possible that some tests are still not started after 1 hour.

Reply:
I think giving it an hour is probably good enough for now.  Anything else is going to be more complicated/have similar problems to what we have in flutter/flutter#43681.

Added the check for older than an hour + a new test.